applications/nrf_desktop/src/modules/led_stream.c

```diff
@@ -0,0 +1,290 @@
+
+	led->rx_idx = next_index(led->rx_idx);
+
+	return true;
+}
+
+static void send_effect(const struct led_effect *effect, struct led *led)
+{
+	LOG_INF("Send effect to leds");
+
+	struct led_event *led_event = new_led_event();
+
+	led_event->led_id = led->id;
+	led_event->led_effect = effect;
+
+	__ASSERT_NO_MSG(led_event->led_effect->steps);
+
+	EVENT_SUBMIT(led_event);
+}
+
+static size_t count_free_places(struct led *led)
+{
+	size_t len = (CONFIG_DESKTOP_LED_STREAM_QUEUE_SIZE + led->rx_idx - led->tx_idx)
+		     % CONFIG_DESKTOP_LED_STREAM_QUEUE_SIZE;
+	return CONFIG_DESKTOP_LED_STREAM_QUEUE_SIZE - len - 1;
+}
+
+static bool store_data(const struct event_dyndata *data, struct led *led)
+{
+	size_t free_places = count_free_places(led);
+
+	if (free_places > 0) {
+		LOG_INF("Insert data on position %" PRIu8
+			", free places %zu", led->rx_idx, free_places);
+
+		if (!queue_data(data, led)) {
+			return false;
+		}
+	} else {
+		LOG_WRN("Queue is full - dropping incoming step");
+	}
+	return true;
+}
+
+static void send_data_from_queue(struct led *led)
+{
+	size_t free_places = count_free_places(led);
+
+	if (free_places < CONFIG_DESKTOP_LED_STREAM_QUEUE_SIZE - 1) {
+		LOG_INF("Sending data");
+
+		led->led_stream_effect.steps = &led->steps_queue[led->tx_idx];
+		led->led_stream_effect.step_count = 1;
+
+		send_effect(&led->led_stream_effect, led);
+
+		led->tx_idx = next_index(led->tx_idx);
+	} else {
+		LOG_INF("No steps ready in queue, sending previous state effect");
+
+		led->streaming = false;
+
+		send_effect(led->state_effect, led);
+	}
+}
+
+static void handle_incoming_step(const struct config_event *event)
+{
+	if (!initialized) {
+		LOG_WRN("Not initialized");
+		return;
+	}
+
+	size_t led_id = event->dyndata.data[LED_ID_POS];
+	struct led *led = &leds[led_id];
+
+	led->id = led_id;
+
+	if (!store_data(&event->dyndata, led)) {
+		return;
+	}
+
+	if (!led->streaming) {
+		LOG_INF("Sending first led effect for led %zu", led_id);
+
+		led->streaming = true;
+
+		send_data_from_queue(led);
+	}
+}
+
+static void handle_config_event(const struct config_event *event)
+{
+	if (!event->store_needed) {
+		/* Accept only events coming from transport. */
+		return;
+	}
+
+	if (GROUP_FIELD_GET(event->id) != EVENT_GROUP_LED_STREAM) {
+		/* Only LED STREAM events. */
+		return;
+	}
+
+	switch (TYPE_FIELD_GET(event->id)) {
+	case STREAM_DATA:
+		handle_incoming_step(event);
+		break;
+
+	default:
+		LOG_WRN("Unknown LED STREAM config event");
+		break;
+	}
+}
+
+static void handle_config_fetch_request_event(
+	const struct config_fetch_request_event *event)
+{
+	LOG_INF("Handle config fetch request event");
+
+	if (GROUP_FIELD_GET(event->id) != EVENT_GROUP_LED_STREAM) {
+		/* Only LED STREAM events. */
+		return;
+	}
+
+	size_t led_id = MOD_FIELD_GET(event->id);
+
+	if (led_id < 0 && led_id >= CONFIG_DESKTOP_LED_COUNT) {
+		LOG_WRN("Wrong LED id");
+		return;
+	}
+
+	struct led *led = &leds[led_id];
+
+	u8_t free_places = count_free_places(led);
+
```

Comment:
`u8_t is_ready = initialized;`

this is to ensure the size of the variable

Reply:
standard only defines it must be larger to store 1 and 0 - size can be anything that can hold it (e.g. 4 bytes)